contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+
+        // Transfer sender output token
+        _outputToken.safeTransfer(msg.sender, outputAmount);
+        // Emit event
+        emit ExchangeRedeem(msg.sender, _setToken, _outputToken, _amountSetToken, outputAmount);
+        // Return output amount
+        return outputAmount;
+    }
+
+    /**
+     * Redeems an exact amount of SetTokens for ETH.
+     * The SetToken must be approved by the sender to this contract.
+     *
+     * @param _setToken             Address of the SetToken being redeemed
+     * @param _amountSetToken       Amount SetTokens to redeem
+     * @param _minEthReceive        Minimum amount of Eth to receive
+     * @param _componentQuotes      The encoded 0x transactions execute
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        uint256 _minEthReceive,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+        uint ethAmount = _sellComponentsForOutputToken(_setToken, _amountSetToken, _componentQuotes, IERC20(WETH), _issuanceModule);
+        require(ethAmount >= _minEthReceive, "ExchangeIssuance: INSUFFICIENT WETH RECEIVED");
+
+        IWETH(WETH).withdraw(ethAmount);
+        (payable(msg.sender)).sendValue(ethAmount);
+
+        emit ExchangeRedeem(msg.sender, _setToken, IERC20(ETH_ADDRESS), _amountSetToken, ethAmount);
+        return ethAmount;
+         
+    }
+    
+    /**
+     * Whitelists an issuance module
+     *
+     * @param _issuanceModule    Struct containing data on issuance module to add
+     */
+    function _addIssuanceModule(address _issuanceModule, bool _isDebtIssuanceModule) internal {
+        allowedIssuanceModules[_issuanceModule].isAllowed = true;
+        allowedIssuanceModules[_issuanceModule].isDebtIssuanceModule = _isDebtIssuanceModule;
+    }
+
+    /**
+     * Removes an issuance module from the whitelist
+     *
+     * @param _issuanceModuleAddress    Address of issuance module to remove
+     */
+    function _removeIssuanceModule(address _issuanceModuleAddress) internal {
+        allowedIssuanceModules[_issuanceModuleAddress].isAllowed = false;
+    }
+
+    /**
+     * Sets a max approval limit for an ERC20 token, provided the current allowance
+     * is less than the required allownce.
+     *
+     * @param _token    Token to approve
+     * @param _spender  Spender address to approve
+     */
+    function _safeApprove(IERC20 _token, address _spender, uint256 _requiredAllowance) internal {
+        uint256 allowance = _token.allowance(address(this), _spender);
+        if (allowance < _requiredAllowance) {
+            _token.safeIncreaseAllowance(_spender, type(uint96).max - allowance);
```

Comment:
Is accidental overflow possible here? `allowance` and `_requiredAllowance` are `uint256` but the max is `96` ?

Reply:
Good point. Again this is copy paste from EI V1. Actually my `git blame` returns @cgewecke as the author of the equivalent lines in those contracts. So not sure who might be able to clear this up.

I guess the answer to this depends in part on the outcome of the discussion of why we are using MAX_UINT96 in the first place.

Reply:
IIRC I am not the actual author (not sure how I got commits on this - perhaps via V2 which I copied V1 for). Noah definitely knows this contract the best of any of us I think.

Reply:
As mentioned above I replaced `type(uint96).max` with `type(uint256).max` so overflow should not be an issue.